avro/src/main/scala/AvroTransformer.scala

```diff
@@ -0,0 +1,39 @@
+package com.memsql.spark.examples.avro
+
+import com.memsql.spark.etl.api.{UserTransformConfig, SimpleByteArrayTransformer}
+import com.memsql.spark.etl.utils.PhaseLogger
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.{DataFrame, Row, SQLContext}
+
+import spray.json.JsValue
+import org.apache.avro.Schema
+import org.apache.avro.generic.GenericData
+import org.apache.avro.io.DecoderFactory
+import org.apache.avro.specific.SpecificDatumReader
+
+class AvroTransformer extends SimpleByteArrayTransformer {
+  def AvroRDDToDataFrame(sqlContext: SQLContext, rdd: RDD[Array[Byte]], schemaJsonString: String): DataFrame = {
+    val rowRDD: RDD[Row] = rdd.map(bytes => {
```

Comment:
You should use rdd.mapPartitions and create a parser and an avroSchema for each partition instead of for each record in this RDD, for efficiency's sake.